src/detect-engine-content-inspection.c

```diff
@@ -207,8 +218,12 @@ static int DetectEngineContentInspectionInternal(DetectEngineThreadCtx *det_ctx,
                 }
 
                 if (cd->flags & DETECT_CONTENT_OFFSET_VAR) {
-                    if (det_ctx->byte_values[cd->offset] > offset)
-                        offset = det_ctx->byte_values[cd->offset];
+                    if (det_ctx->byte_values[cd->offset] > offset) {
+                        if (det_ctx->byte_values[cd->offset] > UINT32_MAX) {
```

Comment:
how would this happen? Can we avoid it from getting set? It seems painful to put all these checks here if we can avoid it

Reply:
I think with a rule using `byte_extract` with 8 bytes..?